diff --git a/cjs/react-refresh-runtime.development.js b/cjs/react-refresh-runtime.development.js
index b0b84c11c419b34866defde9e83a0226f4b72bd4..c1bf017151642fea1166417ae99ea20ab5f519c9 100644
--- a/cjs/react-refresh-runtime.development.js
+++ b/cjs/react-refresh-runtime.development.js
@@ -330,6 +330,10 @@
         if (didError) throw firstError;
         return update;
       } finally {
+        if (window.notifyRefreshComplete) {
+          window.notifyRefreshComplete();
+        }
+
         isPerformingRefresh = !1;
       }
     };
